gnovm/pkg/gnolang/uverse.go

```diff
@@ -710,6 +710,7 @@ func UverseNode() *PackageNode {
 			case *MapType:
 				mv := arg0.TV.V.(*MapValue)
 				mv.DeleteForKey(m.Store, &itv)
+				m.Realm.DidUpdate(mv, nil, nil)
```

Comment:
I think this needs a bit more than just updating the map value -- the deleted object should be marked as well so that it's not hanging around forever.  We can do something similar to how this is done during assignment:
https://github.com/gnolang/gno/blob/8a0de59a6b0ad8cc13e1ebfeefd4ea005dd498d3/gnovm/pkg/gnolang/values.go#L279-L283

In this case it would just need to get the object value from `itv` if the realm isn't nil, and then call `DidUpdate` using that value.  The next block may also need to be changed to handle object deletion from a native map type; I will test it out.

Reply:
```suggestion
if m.Realm != nil {
	// delete value and mark it deleted if it's a object
	oo1 := itv.GetFirstObject(m.Store)
	m.Realm.DidUpdate(mv, oo1, nil)
}
```

How about this?

Reply:
Sorry, what I typed above is slightly incorrect.  `itv` is not a persisted value here -- just a key used to find the persisted value.  What we need is a way to retrieve both the key and the value `TypeValue`s of the element that is being deleted. Then we can get each object and mark it as deleted.  Btw, we need this for the key also because go allows this like comparable structs to be used as keys.

Reply:
So we need to mark both 'key' and 'value' as deleted, am I right??

Reply:
Yes

Reply:
gave my best shot, beec18754dff199a8e9a92f5f02380f58b9d82d8

Reply:
Nice job, thanks!